lib/job.js

```diff
@@ -71,80 +70,65 @@ Job.fromId = function(queue, jobId){
   }
   return queue.client.hgetall(queue.toKey(jobId)).then(function(jobData){
     if(!_.isEmpty(jobData)){
-      return Job.fromData(queue, jobId, jobData);
+      return Job.fromData(queue, jobData);
     }else{
       return null;
     }
   });
 };
 
-Job.prototype.toData = function(){
-  return {
-    name: this.name,
-    data: JSON.stringify(this.data || {}),
-    opts: JSON.stringify(this.opts || {}),
-    progress: this._progress,
-    delay: this.delay,
-    timestamp: this.timestamp,
-    attempts: this.attempts,
-    attemptsMade: this.attemptsMade,
-    failedReason: this.failedReason,
-    stacktrace: JSON.stringify(this.stacktrace || null),
-    returnvalue: JSON.stringify(this.returnvalue || null)
-  };
-};
-
 Job.prototype.progress = function(progress){
   if(progress){
     var _this = this;
     this._progress = progress;
-    return this.queue.client.hset(this.queue.toKey(this.jobId), 'progress', progress).then(function(){
+    return this.queue.client.hset(this.queue.toKey(this.id), 'progress', progress).then(function(){
       _this.queue.distEmit('progress', _this, progress);
     });
   }else{
     return this._progress;
   }
 };
 
-//
-// toData and fromData should be deprecated.
-//
 Job.prototype.toJSON = function(){
   var opts = _.extend({}, this.opts || {});
-  opts.jobId = this.jobId;
+  opts.jobId = this.id;
   return {
     name: this.name,
-    id: this.jobId,
     data: this.data || {},
     opts: opts,
     progress: this._progress,
-    delay: this.delay,
+    delay: this.delay, // Move to opts
     timestamp: this.timestamp,
-    attempts: this.attempts,
     attemptsMade: this.attemptsMade,
     failedReason: this.failedReason,
     stacktrace: this.stacktrace || null,
     returnvalue: this.returnvalue || null
   };
 };
 
+Job.prototype.toData = function(){
+  var json = this.toJSON();
+  json.data = JSON.stringify(json.data);
+  json.opts = JSON.stringify(json.opts);
+  json.stacktrace = JSON.stringify(json.stacktrace);
+  json.returnvalue = JSON.stringify(json.returnvalue);
+
+  return json;
+};
+
 /**
   Return a unique key representing a lock for this Job
 */
 Job.prototype.lockKey = function(){
-  return this.queue.toKey(this.jobId) + ':lock';
+  return this.queue.toKey(this.id) + ':lock';
 };
 
 /**
   Takes a lock for this job so that no other queue worker can process it at the
   same time.
 */
-Job.prototype.takeLock = function(renew, ensureActive){
-  var _this = this;
-  return scripts.takeLock(this.queue, this, renew, ensureActive).then(function(lock) {
-    if (lock){
-      _this.lock = lock;
-    } 
+Job.prototype.takeLock = function(){
```

Comment:
Yeah, would be nice if the unit test implemented it itself. Just to keep the core queue implementation pure and simple.